src/org/zaproxy/zap/extension/treetools/PopupMenuTreeTools.java

```diff
@@ -59,7 +59,7 @@ private void expandOrCollapseFromNode(TreePath parent, boolean expand) {
 		TreeNode tn = (TreeNode) parent.getLastPathComponent();
 		
 		if (tn.getChildCount() > 0) {
-			for (Enumeration<TreeNode> e = tn.children(); e.hasMoreElements();) {
+			for (Enumeration<? extends TreeNode> e = tn.children(); e.hasMoreElements();) {

```

Comment:
Could they have all been done this way to avoid the casting?

Reply:
No, we would still need to cast (in this case it's not important the actual type).

Reply:
K